src/libraries/System.Private.CoreLib/src/System/IO/Strategies/BufferedFileStreamStrategy.cs

```diff
@@ -649,19 +649,19 @@ public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationTo
                 bool releaseTheLock = true;
                 try
                 {
-                    // hot path #1 if the write completely fits into the buffer, we can complete synchronously:
```

Comment:
`ReadAsync` has a very similar couple of hot paths. Would it make sense to get the conditions inverted there too? After asking Stephen's concern with the semaphore, of course.

https://github.com/dotnet/runtime/blob/7878130397bca88d63c5eea7d4a6185e491ee954/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/BufferedFileStreamStrategy.cs#L356-L369

Reply:
@carlossanlop good point, done